src/main/java/com/direwolf20/buildinggadgets/common/tools/NBTTool.java

```diff
@@ -321,4 +322,19 @@ public static NBTTagList createStringList(String[] strings) {
         }
         return toAppendTo;
     }
+
+    /**
+     * If the given stack has a tag, returns it. If the given stack does not have a tag, it will set a reference and return the new tag
+     * compound.
+     */
+    public static NBTTagCompound getOrNewTag(ItemStack stack) {
+        if (stack.hasTagCompound()) {
+            return stack.getTagCompound();
+        } else {
+            NBTTagCompound tag = new NBTTagCompound();
+            stack.setTagCompound(tag);
+            return tag;
```

Comment:
Lines 334-336 are unnecessarily nested in an else clause.

Reply:
What this method does is it returns a new tag if the ItemStack doesn’t have one already, like what the method name and java doc says. I can rename/document it more appropriately or inline it if you think it is unviable.

`ItemStack#hasTagCompund()` checks if the `tagCompound` field is null.

Reply:
Sorry I overlooked the “nested in a else clause”, please ignore the comment

Reply:
You can delete your comments...